### DISCUSSION TITLE
v1.3-experimental1

### DISCUSSION THREAD

**Discussion by gabbybilka** [microsoft/WindowsAppSDK#3406] [Announcements]
## 1.3 Experimental 1 (1.3.230202101-experimental1) 🧪

The 1.3 experimental release of Windows App SDK is now available! Releases in the experimental channel include features that are in the early stages of development. Experimental features may be removed from the next release, or may never be released. See [Windows App SDK release channels](https://docs.microsoft.com/windows/apps/windows-app-sdk/release-channels) for more information about the various release channels.

This release includes bug fixes from 1.2 and provides access to non-stable APIs and features for WinUI 3, Graphics, Content, Dispatching, and Windowing. 

**Check out the [release notes](https://aka.ms/windowsappsdk/1.3/1.3-experimental1/release-notes) for all of the new and updated features, limitations, and known issues.**

### Updating to this version

The 1.3.230202101-experimental1 package is available at: https://www.nuget.org/packages/Microsoft.WindowsAppSDK/1.3.230202101-experimental1

Additionally, for those testing deployment of unpackaged apps, download the Windows App SDK Installers and MSIX packages [here](https://aka.ms/windowsappsdk/1.3/1.3.230202101-experimental1/Microsoft.WindowsAppRuntime.Redist.1.3.zip).

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WindowsAppSDK/releases/tag/v1.3-exp1'>v1.3-experimental1</a>.</em>

**Comment by kmgallahan** [comment]
Are any [Additional Experimental APIs](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/experimental-channel#additional-experimental-apis) documented anywhere, or are there plans to add experimental builds to the [API reference](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui?view=windows-app-sdk-1.2)?

**Reply by gabbybilka** [reply]
For the first question: No, as they are experimental APIs we don't plan to ship in 1.3, there are no docs for them yet. Wanted to call them out explicitly so folks knew they were still under construction in comparison to experimental APIs we are targeting to release in 1.3.0.

For the second question, there are no current plans to add experimental builds to the API reference. @MikeHillberg and/or @stevewhims to add additional context.

**Reply by stevewhims** [reply]
Hi, @kmgallahan! On learn.microsoft.com specifically, we document Stable and Release APIs; and that's the continuing plan of record.

Thanks!
-Steve

**Comment by bogdan-patraucean** [comment]
Something is wrong with this sentence:

> Replacing several lines of boilerplate code, you're now able to use Window APIs directly from an AppWindow through AppWindow.Window

1. There is no documentation to reflect the changes
2. I think it's Windows.AppWindow, not the other way arround.
3. Updates should also be [here](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/windowing/windowing-overview), or at least, give us some draft documentation until this version is stable.

**Reply by gabbybilka** [reply]
Yep! This PR to fix is here, opened by @JaiganeshKumaran: https://github.com/MicrosoftDocs/windows-dev-docs/pull/4290

Until a preview release, usage info is here: https://github.com/microsoft/microsoft-ui-xaml/blob/main/specs/appwindow-spec.md

**Comment by michael-hawker** [comment]
I tried updating a project using 1.2 stable to 1.3, just updated the package in NuGet Package Exploerer, but then my project won't compile...

```
1>------ Build started: Project: ItemsRepeaterAlignmentRepro.WinAppSDK, Configuration: Debug x64 ------
1>Build started 2/8/2023 2:01:23 PM.
1>Target MarkupCompilePass1:
1>  C:\Users\...\.nuget\packages\microsoft.windowsappsdk\1.3.230202101-experimental1\buildTransitive\Microsoft.UI.Xaml.Markup.Compiler.interop.targets(523,17): error MSB4064: The "PrecompiledHeaderFile" parameter is not supported by the "CompileXaml" task loaded from assembly: Microsoft.UI.Xaml.Markup.Compiler, Version=1.0.0.0, Culture=neutral, PublicKeyToken=de31ebe4ad15742b from the path: C:\Users\...\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\tools\net472\Microsoft.UI.Xaml.Markup.Compiler.dll. Verify that the parameter exists on the task, the <UsingTask> points to the correct assembly, and it is a settable public instance property.
1>  C:\Users\...\.nuget\packages\microsoft.windowsappsdk\1.3.230202101-experimental1\buildTransitive\Microsoft.UI.Xaml.Markup.Compiler.interop.targets(491,9): error MSB4063: The "CompileXaml" task could not be initialized with its input parameters.
1>Done building target "MarkupCompilePass1" in project "ItemsRepeaterAlignmentRepro.WinAppSDK.csproj" -- FAILED.
```

Looks like I had to restart Visual Studio after updating?

**Reply by agneszitte** [reply]
Got the same errors when simply creating a new WinUI Blank app and trying to launch the App.
I needed like @michael-hawker to restart Visual Studio to be able to launch the App.
(Using Microsoft Visual Studio Professional 2022 (64-bit) - Current Version 17.6.4)
![image](https://github.com/microsoft/WindowsAppSDK/assets/16295702/99365423-21cd-4f88-8866-f2504285f2c8)

**Comment by Balkoth** [comment]
Seems like for the new XAML Backdrop APIs to work you have to have both a xaml and a xaml.cs file?

Just doing it like this crashes the application with code 3221226107 (0xc000027b):

```cs
public partial class App : Application
{
  public App()
  {
    this.InitializeComponent();
  }

  protected override void OnLaunched(Microsoft.UI.Xaml.LaunchActivatedEventArgs args)
  {
    m_window = new Window();
    m_window.SystemBackdrop = new DesktopAcrylicBackdrop();
    m_window.Activate();
  }

  private Window m_window;
}
```

**Reply by gabbybilka** [reply]
@DmitriyKomin @codendone ?

**Reply by codendone** [reply]
SystemBackdrop currently requires the window to have some content. Tweaking the code above to set `m_window.Content` will launch and work correctly:
```
            m_window = new Window();
            m_window.Content = new Grid();
            m_window.SystemBackdrop = new DesktopAcrylicBackdrop();
            m_window.Activate();
```
Thanks for reporting. Tracking this issue as [internal bug](http://task.ms/43347736).

**Comment by smj389** [comment]
What is the recommended way to give feedback on the experimental APIs you don't plan to ship in 1.3?
